src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -164,6 +179,23 @@ public void handleServiceChange(
                 (CacheConfiguration) wms.getRemoteResourcesCacheConfiguration().clone());
     }
 
+    private int getTimeoutMillis() {
+        String timeoutStr =
+                GeoServerExtensions.getProperty(GETMAPKVPREQUESTREADER_TIMEOUT_VARIABLE);
+        try {
+            if (timeoutStr != null) {
+                int timeout = Integer.parseInt(timeoutStr);
+                return timeout;
+            }
+        } catch (NumberFormatException ex) {
+            LOGGER.log(
+                    Level.FINE,
+                    "Number format exception trying to parse the connection timeout global variable",
```

Comment:
Basically here you are swallowing the exception and use the default value:

1. This should be warning, not a debug log message. 
2. You log message should be improved, it should log the value provided by the user and log the default value that will be used.
3, Your log message should have a better context, _onnection timeout_ is a bit too generic.

Reply:
Changed with better logging on a common task method